### DISCUSSION TITLE
Guidance on using animations

### DISCUSSION THREAD

**Discussion by schtandard** [Cimbali/pympress#310] [Q&A] [answered]
I want to use animations in my presentations with the help of the `animate` package. #102 makes me think that this is supported, but when I open a PDF containing an animation, it does not play; I always see the first frame. Do I need extra dependencies for this or am I missing anything else?

Here's an MWE for testing.
```latex
\documentclass{beamer}

\usepackage{animate}

\begin{document}

\begin{frame}
  \centering\Huge\bfseries
  \begin{animateinline}[loop, autoplay]{5}
    \multiframe{10}{n=0+1}{\fbox\n}%
  \end{animateinline}
\end{frame}

\end{document}
```
The animation plays in PDF-Xchange Editor and Adobe Acrobat Reader but not in Pympress.

**Comment by Cimbali** [comment, marked as answer]
I’d be surprised animate works, AFAICT it’s using javascript to dive animations which we don’t support. Embedding the image as a gif works out of the box though.

The issue you link to reported a crash related to animate and that was fixed, but there’s no support added for animate at this time.

**Reply by Angelo-max-hub** [reply]
Is it possible to implement a change that allows the user to define a specific range of slides for automatic transitions? Currently, the CLI allows configuring the entire PDF to advance automatically at set intervals, but not just a specific subset of slides.
If we could restrict this to a page range, or isolate configurations to certain pages, we might be able to create animations after compiling to PDF rather than before, during the LaTeX Beamer code where animations rely on JavaScript. In this new scenario, it would be the user's responsibility to inform pympress—via GUI or CLI—of the starting slide number of a movement, the ending slide, and the timing between transitions; pympress would then handle the automatic playback. Both Beamer and Typst's Touying already have commands to duplicate slides multiple times with minor changes. We could leverage the existing CLI definitions for automatic transitions but with the added capability of isolating settings to specific slides.
Theoretically, the user would generate a PDF using LaTeX Beamer or Typst Touying and then use the GUI, CLI, or a configuration file of any extension to set up the automatic transitions—perhaps even a transition loop.
